src/main/java/net/openhft/chronicle/queue/ChronicleHistoryReaderMain.java

```diff
@@ -67,36 +91,66 @@ protected void setup(@NotNull final CommandLine commandLine, @NotNull final Chro
             chronicleHistoryReader.withSummaryOutput(Integer.parseInt(commandLine.getOptionValue('u')));
     }
 
+    /**
+     * Initializes a new instance of {@link ChronicleHistoryReader}.
+     *
+     * @return A new {@link ChronicleHistoryReader} instance
+     */
     @NotNull
     protected ChronicleHistoryReader chronicleHistoryReader() {
         return new ChronicleHistoryReader();
     }
 
+    /**
+     * Parses command-line arguments using Apache Commons CLI.
+     * If help is requested, it prints the help message and exits.
+     *
+     * @param args    Command-line arguments
+     * @param options Available command-line options
+     * @return Parsed {@link CommandLine} object
+     */
     protected CommandLine parseCommandLine(@NotNull final String[] args, final Options options) {
-        final CommandLineParser parser = new DefaultParser();
+        final CommandLineParser parser = new DefaultParser(); // Initialize command-line parser
         CommandLine commandLine = null;
+
         try {
-            commandLine = parser.parse(options, args);
+            commandLine = parser.parse(options, args); // Parse arguments
 
+            // If help option is selected, print help and exit
             if (commandLine.hasOption('h')) {
                 printHelpAndExit(options, 0);
             }
         } catch (ParseException e) {
-            printHelpAndExit(options, 1, e.getMessage());
+            // If parsing fails, print help with an error message and exit
+            String cmdLine = Arrays.toString(args);
+            printHelpAndExit(options, "[-h]".equals(cmdLine) ? 0 : 1, e.getMessage());
         }
 
         return commandLine;
     }
 
+    /**
+     * Prints help and exits the program.
+     *
+     * @param options Command-line options
+     * @param status  Exit status
+     */
     protected void printHelpAndExit(final Options options, int status) {
         printHelpAndExit(options, status, null);
     }
 
+    /**
+     * Prints help and exits the program, optionally with a message.
+     *
+     * @param options Command-line options
+     * @param status  Exit status
+     * @param message Optional message to print before help
+     */
     protected void printHelpAndExit(final Options options, int status, String message) {
         final PrintWriter writer = new PrintWriter(System.out);
         new HelpFormatter().printHelp(
                 writer,
-                180,
+                180, // Line width for formatting help output
```

Comment:
Consider extracting to constant and move the javadoc to that constant. Suggest constant name: `HELP_OUTPUT_LINE_WIDTH`